lib/perfdata/influxdbchecktask.cpp

```diff
@@ -0,0 +1,149 @@
+
+	double pendingDataBuffersCrit = MacroProcessor::ResolveMacros("$influxdbwriter_pending_queries_critical$", resolvers, checkable->GetLastCheckResult(),
+		&missingPendingDataBuffersCrit, MacroProcessor::EscapeCallback(), resolvedMacros, useResolvedMacros);
+
+	if (resolvedMacros && !useResolvedMacros)
+		return;
+
+	if (influxdbWriterName.IsEmpty()) {
+		ReportInfluxdbCheck(checkable, commandObj, cr, "Attribute 'influxdbwriter_name' must be set.");
+		return;
+	}
+
+	InfluxdbWriter::Ptr influxdbWriter = InfluxdbWriter::GetByName(influxdbWriterName);
+
+	if (!influxdbWriter) {
+		ReportInfluxdbCheck(checkable, commandObj, cr, "InfluxdbWriter '" + influxdbWriterName + "' object does not exist.");
+		return;
+	}
+
+	if (influxdbWriter->IsPaused()) {
+		ReportInfluxdbCheck(checkable, commandObj, cr, "InfluxdbWriter is temporarily disabled on this cluster instance.", state);
+		return;
+	}
+
+	auto errorMessage = influxdbWriter->GetLastErrorMessage();
+	if (!errorMessage.IsEmpty()) {
+		ReportInfluxdbCheck(checkable, commandObj, cr, errorMessage, ServiceCritical);
+		return;
+	}
+
+	double wqItemPerSec = influxdbWriter->GetQueryCount(60) / 60.0;
+	double pendingDataBuffers = influxdbWriter->GetPendingDataBuffers();
+	std::ostringstream msgBuffer;
+
+	msgBuffer << "Queries per second: " << std::fixed << std::setprecision(3) << wqItemPerSec << ".";
+
+	/* Check whether the thresholds have been defined and match. */
+	if (missingQueriesCritical.IsEmpty() && wqItemPerSec < queriesCritical) {
+		msgBuffer << " " << wqItemPerSec << " queries/s lower than critical threshold (" << queriesCritical << " queries/s).";
+
+		state = ServiceCritical;
+	} else if (missingQueriesWarning.IsEmpty() && wqItemPerSec < queriesWarning) {
+		msgBuffer << " " << wqItemPerSec << " queries/s lower than warning threshold (" << queriesWarning << " queries/s).";
+
+		state = ServiceWarning;
+	}
+
+	if (missingPendingDataBuffersCrit.IsEmpty() && pendingDataBuffers > pendingDataBuffersCrit) {
+		msgBuffer << " " << pendingDataBuffers << " pending data buffers greater than critical threshold ("
+			<< pendingDataBuffersCrit << " queries).";
+
+		state = ServiceCritical;
+	} else if (missingPendingDataBuffersWarn.IsEmpty() && pendingDataBuffers > pendingDataBuffersWarn) {
+		msgBuffer << " " << pendingDataBuffers << " pending data buffers greater than warning threshold ("
+			<< pendingDataBuffersWarn << " queries).";
+
+		state = ServiceWarning;
+	}
+
+	cr->SetPerformanceData(new Array({
+		{ new PerfdataValue("queries", wqItemPerSec, false, "", queriesWarning, queriesCritical) },
+		{ new PerfdataValue("queries_1min", influxdbWriter->GetQueryCount(60)) },
+		{ new PerfdataValue("queries_5mins", influxdbWriter->GetQueryCount(5 * 60)) },
+		{ new PerfdataValue("queries_15mins", influxdbWriter->GetQueryCount(15 * 60)) },
+		{ new PerfdataValue("pending_datas", pendingDataBuffers, false, "", pendingDataBuffersWarn, pendingDataBuffersCrit) }
+	}));
```

Comment:
@Al2Klimov  You want to change this in the IDO check in #8401 and this is heavily inspired by the IDO plugin. Since I haven't taken the time to look over that PR in detail, is this a change we'd want here too?

Reply:
Let’s come back to this construction area once one of the two PRs has been merged.